python/delta_sharing/reader.py

```diff
@@ -105,30 +121,48 @@ def _copy(
 
     @staticmethod
     def _to_pandas(
-        add_file: AddFile, converters: Dict[str, Callable[[str], Any]], limit: Optional[int]
+        action: FileAction, converters: Dict[str, Callable[[str], Any]], limit: Optional[int]
     ) -> pd.DataFrame:
-        url = urlparse(add_file.url)
+        url = urlparse(action.url)
         if "storage.googleapis.com" in (url.netloc.lower()):
             # Apply the yarl patch for GCS pre-signed urls
             import delta_sharing._yarl_patch  # noqa: F401
 
         protocol = url.scheme
         filesystem = fsspec.filesystem(protocol)
 
-        pa_dataset = dataset(source=add_file.url, format="parquet", filesystem=filesystem)
+        pa_dataset = dataset(source=action.url, format="parquet", filesystem=filesystem)
         pa_table = pa_dataset.head(limit) if limit is not None else pa_dataset.to_table()
         pdf = pa_table.to_pandas(
             date_as_object=True, use_threads=False, split_blocks=True, self_destruct=True
         )
 
         for col, converter in converters.items():
             if col not in pdf.columns:
-                if col in add_file.partition_values:
+                if col in action.partition_values:
                     if converter is not None:
-                        pdf[col] = converter(add_file.partition_values[col])
+                        pdf[col] = converter(action.partition_values[col])
                     else:
                         raise ValueError("Cannot partition on binary or complex columns")
                 else:
                     pdf[col] = None
 
+        # If available, add timestamp and version columns from the action.
+        # All rows of the dataframe will get the same value.
+        if action.timestamp is not None:
+            assert DeltaSharingReader._current_timestamp_col_name() not in pdf.columns
+            pdf[DeltaSharingReader._current_timestamp_col_name()] = action.timestamp
```

Comment:
And maybe we should convert the `long` timestamp into string format.

Reply:
As discussed, keep it in the same format as input.
The client can always process the column and convert it as needed.

Reply:
sg, let's revisit when necessary.